e3sm_diags/run.py

```diff
@@ -254,4 +255,7 @@ def _get_other_diags(self, run_type):
         return params
 
 
-runner = Run()
+try:
+    runner = Run()
+except Exception:
+    logger.exception("Error traceback:", exc_info=True)
```

Comment:
If e3sm diags crashes, the final traceback is captured here before the session ends.
Note, the file is not moved to the provenance directory if this happens.

Reply:
The logger now captures the final traceback if e3sm diags crashes and moves the log file to the prov directory.